File: halcyon-web/src/main/resources/static/js/service/workflow/leavebill/leaveBillService.js
/**
 * author:Marshal
 * description:service层,与后端交互
 */
app.service("leaveBillService", function ($http) {

    this.createLeaveBillProcessInstance = function (actBizLeave) {
        return $http.post(baseContextPath + "/workflow/leaveBill/createNewInstance", actBizLeave);
    };

    this.queryById = function (id) {
        return $http.post(baseContextPath + "/workflow/leaveBill/queryById/" + id);
    };

});

File: halcyon-workflow/src/main/java/com/marshal/halcyon/workflow/leavebill/service/LeaveBillService.java
package com.marshal.halcyon.workflow.leavebill.service;

import com.marshal.halcyon.workflow.leavebill.entity.ActBizLeave;

import java.util.List;

public interface LeaveBillService {

    void createLeaveBillInstance(ActBizLeave actBizLeave);

    List<ActBizLeave> query(ActBizLeave condition);

    List<ActBizLeave> submit(List<ActBizLeave> actBizLeaves);

    List<ActBizLeave> remove(List<ActBizLeave> actBizLeaves);

    ActBizLeave queryById(Long id);
}
